### PR TITLE
fix: Remove database requirements for Property Entrance Component and greatly simplify logic

### PR DESCRIPTION
**Pull request by EmosewaMC**
Tested that the interface is identical in behavior to before
best friends get a special icon
alt accounts get a special icon
friends get a different special icon
arrows to go forward only show up if there are more properties to view
search looks in all 3 string fields
having an exact number of properties to fill a screen (12) does not show an arrow

### CODE REVIEW

**Review by aronwk-aaron — APPROVED** [review]

File: dCommon/dEnums/ePropertySortType.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#ifndef EPROPERTYSORTTYPE_H
+#define EPROPERTYSORTTYPE_H
```
Comment (jadebenn) [inline dCommon/dEnums/ePropertySortType.h:2]: I think you told me this earlier, but why aren't we just using `#pragma once` here?

Comment (EmosewaMC) [reply dCommon/dEnums/ePropertySortType.h:2]: doesn't work with pre-compiled headers on gcc

File: dGame/dComponents/PropertyEntranceComponent.h [hunk 2]
```diff
@@ -59,10 +59,6 @@ class PropertyEntranceComponent final : public Component {
 	 */
 	[[nodiscard]] LWOMAPID GetMapID() const { return m_MapID; };
```
Comment (jadebenn) [inline dGame/dComponents/PropertyEntranceComponent.h:60]:
```suggestion
	[[nodiscard]] LWOMAPID GetMapID() const noexcept { return m_MapID; };
```
the nitpickiest of nitpicks: a simple getter for a basic integer type can be noexcept

Comment (EmosewaMC) [reply dGame/dComponents/PropertyEntranceComponent.h]: but what if

File: dGame/User.h [hunk 3]
```diff
@@ -25,7 +25,7 @@ class User {
 	User& operator=(const User& other);
 	bool operator==(const User& other) const;
 
-	uint32_t GetAccountID() { return m_AccountID; }
+	uint32_t GetAccountID() const { return m_AccountID; }
```
Comment (jadebenn) [inline dGame/User.h:28]:
```suggestion
	uint32_t GetAccountID() const noexcept { return m_AccountID; }
```
ditto

**Review by jadebenn — APPROVED** [review]
nitpicks aside, built and ran and it looks good to me!

**Review by jadebenn — APPROVED** [review]

**Review by aronwk-aaron — APPROVED** [review]

**Pull request merged**